### ISSUE TITLE
After downloading 1000 images from Sankaku, I can only download in batches of 20 at a time before it starts skipping

### ISSUE THREAD

**Issue by sugnimmingus** [Nandaka/DanbooruDownloader#225]
I've already added my username and password hash to the cookie login. When I try to start a new batch job from, say, page 53, it downloads 20 images, pauses for a sec, and then just skips the rest. I can then make a new batch job at page 54 for example, and it will correctly download the next 20 images from that page, before skipping the rest again. 

What's going on here? Why is it skipping everything? There was no skipping for the first 1000 images (50 pages) and as soon as it arrived at page 51, the skipping began. 

How do I fix this?

**Comment by sugnimmingus** [comment]
I went in to the logs to get a better look at what was going on, and what I found is weird. After downloading the 20 images of a page, when it (correctly) increments to the next page, even though the URL is correct for the next page number, it somehow finds and tries to download the 20 images of page 1. It will continue to do this for each successive page. And since the images from page 1 already exist in my directory, it just skips them, and the cycle repeats. I don't understand why it would be finding the images from page 1, when the URL is correctly labeled as page 143 for example.

**Comment by Nandaka** [comment]
looks like sankaku doesn't use the standard api, as it depends on additional &next= parameter (if you check from browser and disable auto page.

**Comment by Nandaka** [comment]
try latest releases from https://github.com/Nandaka/DanbooruDownloader/releases
ensure the cookies is updated.

**Comment by sugnimmingus** [comment]
Strange. Since updating, I'm getting this error when I try to run the same search query I was running last night. 
![image](https://user-images.githubusercontent.com/73413785/97112824-e06f8600-16bc-11eb-8ba2-2ae538ff80c6.png)

The search query is: sex -3d -animated order:quality height:512..1000
This query was not giving me any issues last night. Any ideas what's going on?

**Comment by Nandaka** [comment]
looks like it failed to find a match to resolve the next parameter

**Comment by Nandaka** [comment]
yeah, they changed the pattern again. Last time it was the next post id, now looks like it got decimal in it
![image](https://user-images.githubusercontent.com/836867/97113616-99b67600-1726-11eb-9fa8-bd4f811612b3.png)

**Comment by sugnimmingus** [comment]
What am I supposed to do with this information? And why would that only apply to order:Quality?

**Comment by Nandaka** [comment]
no idea, sankaku is not using standard danbooru api. You can try to use https://bionus.github.io/imgbrd-grabber/ as it more mature/maintained.

**Comment by Nandaka** [comment]
updated exe, just replace the old one
[DanbooruDownloader3.exe.zip](https://github.com/Nandaka/DanbooruDownloader/files/5439465/DanbooruDownloader3.exe.zip)

**Comment by yami-no-tusbas** [comment]
btw, Grabber by bionus doesn't support Sankaku complex AFAIK, sankaku admin always do it's best to oppose to grabbers programs...

**Comment by sugnimmingus** [comment]
Looks like your update fixed my problem! Thanks for the help! Will let you know if I have any future errors.

**Comment by sugnimmingus** [comment]
Nevermind I take it back, it's still looping the same posts after downloading 500 images for some reason, which is less than the original 1000. Any ideas? It seems to have begun this time after reaching page 26. However, interestingly enough resuming from page 26 works and then it begins skipping at page 27. This is the same thing that was happening before except for page 51+. I wonder what's going on...

**Comment by sugnimmingus** [comment]
I'm not sure what you're recommending I do. What's strange is I successfully tested page 51, and it correctly proceeded to page 52 without skipping images. But after downloading pages 1-25, it no longer works. I'm really not sure what you're saying I need to do. I understand that the API is unusual, but if I manually visit the website, I can correctly load results from any page. So how do I get the loader to do this? I don't believe it's impossible.

**Comment by sugnimmingus** [comment]
Okay so I've done some more research. It has to do with the fact that, on Sankaku, you can't edit the page number in the URL if the URL contains ?next. Is there anyway for you to code it in such a way that it drops the ?next part and just inputs the page number for each page?

**Comment by sugnimmingus** [comment]
One person at Sankaku recommended using this new beta API https://capi-v2.sankakucomplex.com 
Is this possible?

**Comment by sugnimmingus** [comment]
Any chance you'd be able to address this?

**Comment by poopinaloop** [comment]
I'm having the same issue. is there any way to make the program ignore the "&next" part of the url? thanks!
